### PR TITLE
Avoid lambda permission for log subscriptions from depending on itself

### LINKED ISSUE #11869: CloudWatch Logs Lambda Permissions create circular dependency
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

Serverless version 3.29.0 included the merge from https://github.com/serverless/serverless/pull/11788 with changes to:

`lib/plugins/aws/package/compile/events/cloud-watch-log.js`
On line 92 this was included: https://github.com/serverless/serverless/pull/11788/files#diff-7947b4d174bc2f0b06b35408b1d01f10e7fed97f903d028e0f97bcdd94d8a083R92

```(nodejs)
            const dependsOn = [
              lambdaPermissionLogicalId,
              _.get(functionObj.targetAlias, 'logicalId'),
            ].filter(Boolean);
```

Which is later used here:
https://github.com/serverless/serverless/pull/11788/files#diff-7947b4d174bc2f0b06b35408b1d01f10e7fed97f903d028e0f97bcdd94d8a083R115

```(nodejs)
           const permissionTemplate = `
            {
              "Type": "AWS::Lambda::Permission",
              ${dependsOn.length ? `"DependsOn": ${JSON.stringify(dependsOn)},` : ''}
```

(for clarity I am including the value generated for lambdaPermissionLogicalId:
```(nodejs)
        const lambdaPermissionLogicalId =
          this.provider.naming.getLambdaIotFleetProvisioningPermissionLogicalId(functionName);
```
)

This causes a `DependsOn` array to be introduced to the `*LambdaPermissionLogsSubscriptionFilterCloudWatchLog` `AWS::Lambda::Permission` resource with a dependency on itself.

Example from the CloudFormation generated:
```
"FunctionTestLambdaPermissionLogsSubscriptionFilterCloudWatchLog": {
"Type": "AWS::Lambda::Permission",
"DependsOn": [
"FunctionTestLambdaPermissionLogsSubscriptionFilterCloudWatchLog"
 ],
```

While this has been included because the CloudWatchLogsRules require the `DependsOn` (see https://github.com/serverless/serverless/blob/16dd286a1d8aa764003565bcd6d7334be290e9bf/lib/plugins/aws/package/compile/events/cloud-watch-log.js#L99 ), this `DependsOn` for the `permissionTemplate` should not includes the `lambdaPermissionLogicalId`, as that creates a circular reference to itself.

This causes deployments using 3.29.0 to fail during CloudFormation template upload and validation, preventing stack deployment.

### Service configuration (serverless.yml) content

```yaml
N/A
```

### Command name and used flags

N/A

### Command output

```shell
N/A
```

### Environment information

```shell
N/A
```

### PR DESCRIPTION
**Pull request by mjdickinson**
Closes #11869 

I just filtered the resources's own logical ID from the list of dependencies in the lambda permission template.

### CODE REVIEW

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @mjdickinson !

It looks as regression introduced by https://github.com/serverless/serverless/pull/11788 (cc @rnielsen )

Can you add a regression test for it?

**Comment by mjdickinson** [comment]
Tests added.

File: test/unit/lib/plugins/aws/package/compile/events/cloud-watch-log.test.js [hunk 1]
```diff
@@ -46,10 +46,18 @@ describe('AwsCompileCloudWatchLogEvents', () => {
         awsCompileCloudWatchLogEvents.serverless.service.provider.compiledCloudFormationTemplate
           .Resources.FirstLogsSubscriptionFilterCloudWatchLog1.Type
       ).to.equal('AWS::Logs::SubscriptionFilter');
+      expect(
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/cloud-watch-log.test.js:49]: New tests should be configured via `runServerless` util, as e.g. tests in this block: https://github.com/serverless/serverless/blob/5d1c3f7ed4d481619d412b6b453d6eaf94b32b4a/test/unit/lib/plugins/aws/package/compile/events/websockets/index.test.js#L116-L258

See also: https://github.com/serverless/serverless/tree/main/test#unit-tests

In this test file let's add a new `describe` block, that will host those new tests

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @mjdickinson , it looks great, just tests needs to be configured with different method. Let me know if you have any further questions on that, I'll do my best to help

Also after PR is ready for re-review simply re-request review in reviewers box

**Comment by mjdickinson** [comment]
I don't have time to refactor the existing tests.  This PR is an emergency bug with regression tests added, as requested.  Personally, I'm not blocked - I have a workaround.  Please feel free to close the PR.

**Pull request closed without merging**